bindgen/clang.rs

```diff
@@ -973,6 +974,55 @@ impl Cursor {
     pub(crate) fn is_inline_namespace(&self) -> bool {
         unsafe { clang_Cursor_isInlineNamespace(self.x) != 0 }
     }
+
+    /// Obtain the number base (radix) of an integer literal definition corresponding to the cursor.
+    ///
+    /// Returns `None` if unable to infer a base.
+    pub(crate) fn get_literal_radix(&self) -> Option<LiteralRadix> {
+        self.tokens().iter().find_map(|token| {
+            if token.kind == CXToken_Literal {
+                LiteralRadix::from_integer_literal_token(token.spelling())
+            } else {
+                None
+            }
+        })
+    }
+
+    /// Obtain the number base (radix) of an integer literal definition corresponding to the cursor,
+    /// ensuring that the radix is from the literal following a given identifier in the list of
+    /// tokens.
+    ///
+    /// Returns `None` if unable to infer a base.
+    pub(crate) fn get_literal_radix_of_identifier(
+        &self,
+        identifier: &str,
+    ) -> Option<LiteralRadix> {
+        self.tokens()
+            .iter()
+            .scan(false, |identifier_found, token| {
+                if *identifier_found {
+                    Some((true, token))
+                } else if token.kind == CXToken_Identifier {
+                    if token.spelling() == identifier.as_bytes() {
+                        *identifier_found = true;
+                        Some((*identifier_found, token))
+                    } else {
+                        Some((*identifier_found, token))
+                    }
+                } else {
+                    Some((*identifier_found, token))
+                }
```

Comment:
I should simplify the branches and just return `Some((*identifier_found, token))` at the end of the closure instead of this.